### ISSUE TITLE
Rendering a diagram took too many iterations when inContextOf multiple elements

### ISSUE THREAD

**Issue by RedAlbatross** [jonasthewolf/gsn2x#313]
**Describe the bug**
When trying to recreate an example GSN diagram from the GSN Community Standard Version 3 (Figure 0:4-7 An Example Goal Structure), this is the output:
"Rendering a diagram took too many iterations (882). See README.md for hints how to solve this situation."

A SVG is still produced, however its resolution is very large (viewBox="0 0 629116 609"). The y coordinates are reasonable, but half the x coordinates are 0-1000 and half are 626000-628000 range. 

**To Reproduce**
This is a minimal YAML that reproduces the example:

```yaml
G1:
 text: System is acceptably safe to operate.
 classes: [none]
 supportedBy: [G2,  G3,]
G2:
 text: All identified hazards have been eliminated or sufficiently mitigated.
 classes: [none]
 supportedBy: [S1,]
G2.1:
 text: Hazard H1 has been eliminated.
 classes: [none]
 supportedBy: [Sn1,]
G2.2:
 text: Probability of Hazard H2 occuring less than 1x10-6 per year.
 classes: [none]
 supportedBy: [Sn2,]
G2.3:
 text: Probability of Hazard H3 occuring less than 1x10-3 per year.
 classes: [none]
 supportedBy: [Sn2,]
G3:
 text: Software in the Control System has been developed to SIL appropriate to hazards involved.
 classes: [none]
 supportedBy: [S2,]
G3.1:
 text: Primary Protection System Developed to SIL 4.
 classes: [none]
 supportedBy: [Sn3,]
G3.2:
 text: Secondary Protection System Development to SIL2.
 classes: [none]
 supportedBy: [Sn4,]
Sn1:
 text: Formal Verification.
 classes: [none]
Sn2:
 text: Fault Tree Analysis.
 classes: [none]
Sn3:
 text: Process Evidence for SIL4.
 classes: [none]
Sn4:
 text: Process Evidence for SIL2.
 classes: [none]
S1:
 text: Argument over each identified hazard.
 classes: [none]
 supportedBy: [G2.1,  G2.2,  G2.3,]
S2:
 text: Argument over allocated SIL for Primary and Secondary elements.
 classes: [none]
 supportedBy: [G3.1,  G3.2,]
```

**Expected behavior**
A GSN diagram with normal coordinates and dimensions. 

**Additional context**
I have been able to produce many other GSN diagrams using gsn2x. This is the first time I have encountered this problem.

If you remove the context C4 (which is the only element that is inContextOf two other elements), the problem is resolved. Alternatively, removing all context/justification/assumptions apart from C4 also resolves the issue.

**Comment by jonasthewolf** [comment]
Bad news, my computer broke this week. I won't make it this weekend.
I will have a look into your problem as soon as possible though.

**Comment by RedAlbatross** [comment]
@jonasthewolf Sorry to hear that! Many thanks.

**Comment by jonasthewolf** [comment]
@RedAlbatross could you please verify the version of gsn2x you are using. I tried your example above with 2.8.2 and it worked perfectly.

**Comment by RedAlbatross** [comment]
Apologies, I forgot to include the whole yaml file, including the contexts. I am using 2.8.2.
```yaml
G1:
 text: System is acceptably safe to operate.
 classes: [none]
 supportedBy: [G2,  G3,]
 inContextOf: [C1, C2]
G2:
 text: All identified hazards have been eliminated or sufficiently mitigated.
 classes: [none]
 supportedBy: [S1,]
 inContextOf: [C3, C4]
G2.1:
 text: Hazard H1 has been eliminated.
 classes: [none]
 supportedBy: [Sn1,]
G2.2:
 text: Probability of Hazard H2 occuring &lt; 1x10-6 per year.
 classes: [none]
 supportedBy: [Sn2,]
G2.3:
 text: Probability of Hazard H3 occuring &lt; 1x10-3 per year.
 classes: [none]
 supportedBy: [Sn2,]
G3:
 text: Software in the Control System has been developed to SIL appropriate to hazards involved.
 classes: [none]
 supportedBy: [S2,]
 inContextOf: [C4, C5]
G3.1:
 text: Primary Protection System Developed to SIL 4.
 classes: [none]
 supportedBy: [Sn3,]
G3.2:
 text: Secondary Protection System Development to SIL2.
 classes: [none]
 supportedBy: [Sn4,]
Sn1:
 text: Formal Verification.
 classes: [none]
Sn2:
 text: Fault Tree Analysis.
 classes: [none]
Sn3:
 text: Process Evidence for SIL4.
 classes: [none]
Sn4:
 text: Process Evidence for SIL2.
 classes: [none]
S1:
 text: Argument over each identified hazard.
 classes: [none]
 supportedBy: [G2.1,  G2.2,  G2.3,]
S2:
 text: Argument over allocated SIL for Primary and Secondary elements.
 classes: [none]
 supportedBy: [G3.1,  G3.2,]
 inContextOf: [J1, C6]
J1:
 text: SIL apportionment is correct and complete.
 classes: [none]
C1:
 text: Operating Role and Context.
 classes: [none]
C2:
 text:  System Definition.
 classes: [none]
C3:
 text: Tolerability targets (Ref Z).
 classes: [none]
C4:
 text: Hazards identified from FHA (Ref Y).
 classes: [none]
C5:
 text: SIL Guidelines and Processes.
 classes: [none]
C6:
 text: Identified software hazards.
 classes: [none]
```

**Comment by jonasthewolf** [comment]
@RedAlbatross thanks for the hint. I was afraid it was not that easy. I'll have a look at it.

**Comment by jonasthewolf** [comment]
@RedAlbatross I just released 2.8.3 which should fix that problem.
Could you please check if  any of your other examples break?
I would also be very interested in extending my test suite with your examples from the GSN standard. Would you be willing to share them?

**Comment by RedAlbatross** [comment]
@jonasthewolf Many thanks, it works now. This was actually the first example from the standard that I tested. But I will share if I make any of the other examples, which I plan to.
